sdks/java/io/xml/src/test/java/org/apache/beam/sdk/io/xml/XmlIOIT.java

```diff
@@ -0,0 +1,183 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.xml;
+
+import static org.apache.beam.sdk.io.common.IOTestHelper.appendTimestampSuffix;
+
+import com.google.common.collect.ImmutableMap;
+import java.nio.charset.Charset;
+import java.util.Map;
+import org.apache.beam.sdk.io.FileIO;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.common.DeleteFileFn;
+import org.apache.beam.sdk.io.common.HashingFn;
+import org.apache.beam.sdk.io.common.IOTestHelper;
+import org.apache.beam.sdk.io.common.IOTestPipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.Reshuffle;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.transforms.Values;
+import org.apache.beam.sdk.transforms.View;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Integration tests for {@link org.apache.beam.sdk.io.xml.XmlIO}.
+ *
+ * <p>Run those tests using the command below. Pass in connection information via PipelineOptions:
+ * <pre>
+ *  mvn -e -Pio-it verify -pl sdks/java/io/xml
+ *  -Dit.test=org.apache.beam.sdk.io.xml.XmlIOIT
+ *  -DintegrationTestPipelineOptions='[
+ *  "--numberOfRecords=100000",
+ *  "--filenamePrefix=output_file_path",
+ *  "--charset=UTF-8",
+ *  ]'
+ * </pre>
+ * </p>
+ */
+@RunWith(JUnit4.class)
+public class XmlIOIT {
+
+  private static final Map<Integer, String> EXPECTED_HASHES = ImmutableMap.of(
+    1000, "7f51adaf701441ee83459a3f705c1b86",
+    100_000, "af7775de90d0b0c8bbc36273fbca26fe",
+    100_000_000, "bfee52b33aa1552b9c1bfa8bcc41ae80"
+  );
+
+  private static Integer numberOfRecords;
+
+  private static String filenamePrefix;
+
+  private static Charset charset;
+
+  @Rule
+  public TestPipeline pipeline = TestPipeline.create();
+
+  @Rule
+  public TestPipeline readPipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void setup() {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    IOTestPipelineOptions options = TestPipeline
+      .testingPipelineOptions()
+      .as(IOTestPipelineOptions.class);
+
+    filenamePrefix = appendTimestampSuffix(options.getFilenamePrefix());
+    numberOfRecords = options.getNumberOfRecords();
+    charset = Charset.forName(options.getCharset());
+  }
+
+  @Test
+  public void writeThenReadViaSinkAndReadFiles() {
```

Comment:
I think adding a _ReadAll_ transform is a good idea. IMO, it would be even better if a _Write_ transform returning output filenames is added too. I can add those. Should I do this in this PR or maybe finish this one first and then add the transforms? It looks like a separate JIRA issue to me. WDYT?